### PR TITLE
Modified to Android 12 Splash Screen

### PR DESCRIPTION
**Pull request by rohanbabbar04**
Changed the minimum SDK To 21 as it is required for using Android 12 Splash Screen.
Added Implementation for multidex and spotbugs as they were required.

### CODE REVIEW

**Comment by rohanbabbar04** [comment]
Fixes #407

File: app/build.gradle [hunk 1]
```diff
@@ -16,7 +16,7 @@ android {
 
     defaultConfig {
         applicationId "me.hackerchick.catima"
-        minSdkVersion 19
+        minSdkVersion 21
```
Comment (TheLastProject) [inline app/build.gradle:19]: Hmm, this kinda sucks. Means having to leave 35 users without updates. I agree that we don't seem to have another choice though. Just leaving this comment here to let you know that if you find some clever trick to stop us from needing to drop Android 4.4 support that would be cool.

File: app/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -82,8 +82,14 @@
...
     </style>
 
-    <style name="SplashTheme" parent="@style/AppTheme">
-        <item name="android:windowBackground">@drawable/background_splash</item>
+    <!--    Splash Screen Theme-->
+    <style name="Theme.App.Starting" parent="Theme.SplashScreen">
+
+        <item name="windowSplashScreenBackground">@color/ic_launcher_background</item>
+
+        <item name="windowSplashScreenAnimatedIcon">@drawable/background_splash</item>
```
Comment (TheLastProject) [inline app/src/main/res/values/styles.xml:90]: This looks really bad on my Android 12 emulator somehow (Pixel 4)

Screenshot_1633032023

I think you probably shouldn't use the previous complete splash screen as only the icon, but instead use ic_launcher_foreground here? Then we can probably just delete the background_splash file.

File: app/build.gradle [hunk 3]
```diff
@@ -80,13 +80,16 @@ android {
...
     implementation 'androidx.appcompat:appcompat:1.3.1'
     implementation 'androidx.cardview:cardview:1.0.0'
-    implementation 'androidx.constraintlayout:constraintlayout:2.1.0'
+    implementation 'androidx.constraintlayout:constraintlayout:2.1.1'
     implementation 'androidx.exifinterface:exifinterface:1.3.3'
     implementation 'androidx.preference:preference:1.1.1'
     implementation 'com.google.android.material:material:1.4.0'
     coreLibraryDesugaring 'com.android.tools:desugar_jdk_libs:1.1.5'
+// Splash Screen
+    implementation 'androidx.core:core-splashscreen:1.0.0-alpha01'
```
Comment (TheLastProject) [inline app/build.gradle:92]: "A newer version of androidx.core:core-splashscreen than 1.0.0-alpha01 is available: 1.0.0-alpha02"

Should probably use the new version :)

**Review by TheLastProject — CHANGES_REQUESTED** [review]

**Comment by rohanbabbar04** [comment]
So Now it works with min SDK 19 as well 
Do review

File: app/src/main/res/values/styles.xml (hunk 2 above)
Comment (rohanbabbar04) [reply app/src/main/res/values/styles.xml]: I have deleted the background_splash file

Comment (TheLastProject) [reply app/src/main/res/values/styles.xml]: Looks good on my Android 12 emulator and Android 11 physical device now :)

**Comment by TheLastProject** [comment]
After your change the app just plain crashes on Android 4.4:

```
Error while executing: am start -n "me.hackerchick.catima.debug/protect.card_locker.MainActivity" -a android.intent.action.MAIN -c android.intent.category.LAUNCHER
WARNING: linker: libdvm.so has text relocations. This is wasting memory and is a security risk. Please fix.
Starting: Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] cmp=me.hackerchick.catima.debug/protect.card_locker.MainActivity }
Error type 3
Error: Activity class {me.hackerchick.catima.debug/protect.card_locker.MainActivity} does not exist.

Error while Launching activity
```

So please revert that change.

**Comment by rohanbabbar04** [comment]
I have Reverted the changes 
I think we need to upgrade from Android 4.4
@TheLastProject

**Review by TheLastProject — APPROVED** [review]

**Comment by TheLastProject** [comment]
Yup, it's sad but well, we tried :) At least Catima is pretty stable so Android 4.4 users, even though they won't have an update to the new version, will still have an app that is working fine :)

Does https://hacktoberfest.digitalocean.com/profile show this PR in your list? It was still September  30 in some timezones when you made this PR so please double-check if Hacktoberfest accepts this PR before I merge it (otherwise, please recreate the PR so it counts) :)

**Comment by IllusiveMan196** [comment]
Why not have two app versions? One for Android 5-12 and another for earlier systems (although versions earlier than 4.4 should be relatively uncommon already).

**Comment by TheLastProject** [comment]
Because 4.4 is literally the only pre-5.0 version supported and I don't want to take on the extra maintenance work of having a second version for Android 4.4 support. You are free to fork Catima into "Catima Legacy" or so though if you want to put in the work yourself

**Comment by rohanbabbar04** [comment]
> Yup, it's sad but well, we tried :) At least Catima is pretty stable so Android 4.4 users, even though they won't have an update to the new version, will still have an app that is working fine :)
> 
> Does https://hacktoberfest.digitalocean.com/profile show this PR in your list? It was still September 30 in some timezones when you made this PR so please double-check if Hacktoberfest accepts this PR before I merge it (otherwise, please recreate the PR so it counts) :)

Yes It is showing on my hacktoberfest account 
So u can merge it

**Pull request merged**